src/tree/ContainerAppsBranchDataProvider.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, callWithTelemetryAndErrorHandling, IActionContext } from "@microsoft/vscode-azext-utils";
+import { TreeDataProvider, TreeItem } from "vscode";
+import { ContainerAppsExtResourceBase } from "../resolver/ContainerAppsExtResourceBase";
+import { createTreeItemsWithErrorHandling } from "../utils/createTreeItemsWithErrorHandling";
+import { ContainerAppExtParentTreeItem } from "./ContainerAppExtParentTreeItem";
+import { ContainerAppExtTreeItem } from "./ContainerAppExtTreeItem";
+
+export class ContainerAppsBranchDataProvider implements Partial<TreeDataProvider<ContainerAppsExtResourceBase<unknown>>> {
+    public getTreeItem(treeItem: ContainerAppsExtResourceBase<unknown>): TreeItem {
```

Comment:
Probably needs `collapsibleState` too right?

Reply:
Yeah though right now, this branchProvider functionally doesn't do anything except call `getChildren`. I figured once the RG implementation was done, I'd go back and implement it those required functions.